plugins/web-worker-offloading/tests/test-web-worker-offloading.php

```diff
@@ -0,0 +1,192 @@
+	}
+
+	/**
+	 * Tear down the test case.
+	 */
+	public function tear_down(): void {
+		parent::tear_down();
+		$this->reset_wp_dependencies();
+	}
+
+	/**
+	 * @covers ::wwo_get_configuration
+	 */
+	public function test_wwo_get_configuration(): void {
+		$wp_content_dir        = WP_CONTENT_DIR;
+		$partytown_assets_path = 'web-worker-offloading/build/';
+		$config                = wwo_get_configuration();
+
+		$this->assertArrayHasKey( 'lib', $config );
+		$this->assertArrayHasKey( 'forward', $config );
+		$this->assertEmpty( $config['forward'] );
+		$this->assertIsArray( $config['forward'] );
+		$this->assertStringStartsWith( '/' . basename( $wp_content_dir ), $config['lib'] );
+		$this->assertStringEndsWith( $partytown_assets_path, $config['lib'] );
+
+		// Test `wwo_configuration` filter.
+		add_filter(
+			'wwo_configuration',
+			static function ( $config ) {
+				$config['forward'] = array( 'datalayer.push' );
+				$config['debug']   = true;
+				return $config;
+			}
+		);
+
+		$config = wwo_get_configuration();
+
+		$this->assertArrayHasKey( 'forward', $config );
+		$this->assertArrayHasKey( 'debug', $config );
+		$this->assertNotEmpty( $config['forward'] );
+		$this->assertIsArray( $config['forward'] );
+		$this->assertTrue( $config['debug'] );
+		$this->assertContains( 'datalayer.push', $config['forward'] );
+	}
+
+	/**
+	 * @covers ::wwo_init
+	 */
+	public function test_wwo_init(): void {
+		$this->assertEquals( 10, has_action( 'wp_enqueue_scripts', 'wwo_init' ) );
+
+		// Register scripts.
+		wwo_init();
+
+		$wp_content_dir   = WP_CONTENT_DIR;
+		$partytown_config = wwo_get_configuration();
+		$partytown_lib    = dirname( $wp_content_dir ) . $partytown_config['lib'];
+		$before_data      = wp_scripts()->get_inline_script_data( 'web-worker-offloading', 'before' );
+		$after_data       = wp_scripts()->get_inline_script_data( 'web-worker-offloading', 'after' );
+
+		$this->assertTrue( wp_script_is( 'web-worker-offloading', 'registered' ) );
+		$this->assertNotEmpty( $before_data );
+		$this->assertNotEmpty( $after_data );
+		$this->assertEquals(
+			sprintf( 'window.partytown = %s;', wp_json_encode( $partytown_config ) ),
+			$before_data
+		);
+		$this->assertEquals( file_get_contents( $partytown_lib . 'partytown.js' ), $after_data );
+		$this->assertTrue( wp_script_is( 'web-worker-offloading', 'registered' ) );
+
+		// Ensure that Partytown is enqueued when a script depends on it.
+		wp_enqueue_script( 'partytown-test', 'https://example.com/test.js', array( 'web-worker-offloading' ) );
+
+		$this->assertTrue( wp_script_is( 'web-worker-offloading', 'enqueued' ) );
+		$this->assertTrue( wp_script_is( 'partytown-test', 'enqueued' ) );
+	}
+
+	/**
+	 * Data provider for testing `wwo_update_script_type`.
+	 *
+	 * @return array<string, mixed> Data.
+	 */
+	public static function data_update_script_types(): array {
+		return array(
+			'add-script'                           => array(
+				'setup'          => static function (): void {
```

Comment:
_Tiny_ nit: let's replace `setup` with `set_up`, given that the former is a noun and the latter is a verb (and is used as method name in unit tests).